client/proposalmsgs.go

```diff
@@ -620,6 +609,33 @@ func (VirtualChannelProposal) Type() wire.Type {
 	return wire.VirtualChannelProposal
 }
 
+// ProposalID returns the identifier of this channel proposal.
+func (p VirtualChannelProposal) ProposalID() (propID ProposalID) {
+	hasher := newHasher()
+	if err := perunio.Encode(hasher,
+		p.Base(),
+		p.Proposer,
+		wire.Addresses(p.Peers),
+	); err != nil {
+		log.Panicf("proposal ID base encoding: %v", err)
+	}
+	for i := range p.Parents {
+		if err := perunio.Encode(hasher, p.Parents[i]); err != nil {
+			log.Panicf("proposal ID base encoding: %v", err)
+		}
+	}
+	for i := range p.IndexMaps {
+		for j := range p.IndexMaps {
+			if err := perunio.Encode(hasher, (p.IndexMaps[i][j])); err != nil {
+				log.Panicf("proposal ID base encoding: %v", err)
+			}
+		}
+	}
+
+	copy(propID[:], hasher.Sum(nil))
+	return
+}
```

Comment:
We can reuse the `VirtualChannelProposal.Encode` method:
```suggestion
// ProposalID returns the identifier of this channel proposal.
func (p VirtualChannelProposal) ProposalID() (propID ProposalID) {
	hasher := newHasher()
	if err := perunio.Encode(hasher, p); err != nil {
		log.Panicf("encoding proposal: %v", err)
	}
	copy(propID[:], hasher.Sum(nil))
	return
}
```

Reply:
Yeah, I agree.

Reply:
Updated in f3c6404. Did the same for ledger channel and sub channel proposals, in the following commit: 4576ba2.

Reply:
As it was the only suggestion, I squashed the review commit and rebased the branch onto develop.